### PR TITLE
root: Add variants: dcache and oracle

### PR DESCRIPTION
**Pull request by iarspider**
Notice: veccore variant was added earlier, I have just updated the README to reflect that

### CODE REVIEW

**Comment by iarspider** [comment]
Pinging maintainers @chissg, @HadrienG2, @drbenmorgan and @vvolkl

File: var/spack/repos/builtin/packages/root/package.py [hunk 1]
```diff
@@ -127,6 +129,8 @@ class Root(CMakePackage):
     variant('mysql', default=False)
     variant('opengl', default=True,
             description='Enable OpenGL support')
+    variant('oracle', default=True,
```
Comment (vvolkl) [inline var/spack/repos/builtin/packages/root/package.py]:
```suggestion
    variant('oracle', default=False,
```
I think this is a better default for most users of this recipe

File: var/spack/repos/builtin/packages/root/package.py [hunk 2]
```diff
@@ -370,7 +376,7 @@ def cmake_args(self):
             define_from_variant('cocoa', 'aqua'),
             define('dataframe', True),
             define_from_variant('davix'),
-            define('dcache', False),
+            define_from_variant('dcache'),
```
Comment (vvolkl) [inline var/spack/repos/builtin/packages/root/package.py:379]: There is no cmake flag for oracle?

File: var/spack/repos/builtin/packages/root/package.py [hunk 3]
```diff
@@ -137,6 +141,7 @@ class Root(CMakePackage):
             description='Enable Python ROOT bindings')
     variant('qt4', default=False,
             description='Enable Qt graphics backend')
+    variant('r', default=False,
```
Comment (vvolkl) [inline var/spack/repos/builtin/packages/root/package.py]: This line may be an editing error?

File: var/spack/repos/builtin/packages/root/package.py (hunk 1 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/root/package.py]: I agree. Although by default Oracle support is ON in ROOT: https://github.com/root-project/root/blob/master/cmake/modules/RootBuildOptions.cmake#L151

File: var/spack/repos/builtin/packages/root/package.py (hunk 2 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/root/package.py:379]: Nice catch. Added.

File: var/spack/repos/builtin/packages/root/package.py (hunk 3 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/root/package.py]: Yup.

**Review by vvolkl — DISMISSED** [review]
Have not built yet, but the diff looks good. Thanks @iarspider!

**Review by vvolkl — APPROVED** [review]

**Review by haampie — APPROVED** [review]

**Pull request merged**